xla/service/gpu/nccl_api.h

```diff
@@ -51,7 +51,13 @@ class NcclApi {
   // types (also defined as opaque structs).
   struct NcclComm;
   struct NcclPersistentPlanAllocator;
+
+  #if (NCCL_VERSION_CODE < 21901)
+  // fill in missing data type
+  typedef void NcclRegisteredBuffer;
+  #else
   struct NcclRegisteredBuffer;
```

Comment:
`NcclRegisteredBuffer` is not defined anywhere, it's just a unique type to distinguish opaque pointers